modules/programs/autorandr.nix

```diff
@@ -0,0 +1,200 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.autorandr;
+
+  profileModule = types.submodule {
+    options = {
+      fingerprint = mkOption {
+        type = types.attrsOf types.string;
+        description = ''
+          Output name to EDID mapping.
+          Use <code>autorandr --fingerprint</code> to get current setup values.
+        '';
+        default = {};
+      };
+      config = mkOption {
+        type = types.attrsOf configModule;
+        description = "Per output profile configuration.";
+        default = {};
+      };
+      hooks = mkOption {
+        type = profileHooksModule;
+        description = "Profile hook scripts.";
+        default = {};
+      };
+    };
+  };
+
+  configModule = types.submodule {
+    options = {
+      enable = mkOption {
+        type = types.bool;
+        description = "Whether to enable the output.";
+        default = true;
+      };
+      primary = mkOption {
+        type = types.bool;
+        description = "Whether output should be marked as primary";
+        default = false;
+      };
+      position = mkOption {
+        type = types.string;
+        description = "Output position";
+        default = "";
+        example = "5760x0";
+      };
+      mode = mkOption {
+        type = types.string;
+        description = "Output resolution.";
+        default = "";
+        example = "3840x2160";
+      };
+      rate = mkOption {
+        type = types.string;
+        description = "Output framerate.";
+        default = "";
+        example = "60.00";
+      };
+      gamma = mkOption {
+        type = types.string;
+        description = "Output gamma configuration.";
+        default = "";
+        example = "1.0:0.909:0.833";
+      };
+    };
+  };
+
+  hookType = types.lines;
```

Comment:
Originally I wanted to allow users to use either text script or path to a file by doing something like this:
`hookType = with types; coercedTo lines (s: "${pkgs.writeShellScriptBin "hook" s}/bin/hook") path;`
However, it seems that nix checks if a string is a path by [looking at the first character](https://github.com/NixOS/nixpkgs/blob/2ab846211c52e3d7bb26b0888c9777598e921591/lib/types.nix#L230). Because of this, nix will treat something like this as a path: `"{pkgs.i3}/bin/i3-msg restart"`.

@rycee Maybe you know if there is a good way to overcome this limitation? If not, I think the module is usable as it is, external scripts could be imported via `builtins.readFile`.

Reply:
I'm not sure it is possible. On the other hand I think its quite nice and unsurprising to have it as just a string and using `readFile` to insert a script outside the configuration.